message_definitions/v1.0/ardupilotmega.xml

```diff
@@ -241,6 +249,36 @@
         <param index="6">Reserved, send 0</param>
         <param index="7">Reserved, send 0</param>
       </entry>
+      <entry value="43000" name="MAV_CMD_GUIDED_CHANGE_SPEED">
+        <description>Change flight speed at a given rate. This slews the vehicle at a controllable rate between it's previous speed and the new one. (affects GUIDED only. Outside GUIDED, aircraft ignores these commands. Designed for onboard companion-computer command-and-control, not normally operator/GCS control.)</description>
+        <param index="1" label="speed type" enum="SPEED_TYPE">See SPEED_TYPE enum, airspeed or groundspeed.</param>
+        <param index="2" units="m/s">Target Speed</param>
+        <param index="3" label="speed rate-of-change" units="m/s/s">Acceleration rate, 0 to take effect instantly</param>
+        <param index="4">Empty</param>
+        <param index="5">Empty</param>
+        <param index="6">Empty</param>
+        <param index="7">Empty</param>
+      </entry>
+      <entry value="43001" name="MAV_CMD_GUIDED_CHANGE_ALTITUDE">
```

Comment:
Did you consider using more parameters in the existing `handle_change_alt` message?

Reply:
The decision to define the mavlink using three new packets was originally made by @WickedShell  ( mdb ) , so i can't speak to his specific mindset...but.... my understanding is  that as these packets are also only expected to ever be set from onboard the vehicle by a companion computer that's actively in-control of the vehicle( and which does so by 'streaming' these packets repeatedly during manoeuvres), the use-case for these packets is very different to normal guided packets, and the code goes to quite some lengths to ensure that 'normal guided' is still normal guided for normal users.    Perhaps @WickedShell can comment if this isn't enough for you.

Reply:
I'm happy to use the new msg